blockchain/state/database.go

```diff
@@ -123,7 +123,7 @@ func NewDatabaseWithCache(db database.DBManager, cacheSize int, dataArchivingBlo
 	default:
 		cacheConfig = common.FIFOCacheConfig{CacheSize: codeSizeCacheSize}
 	}
-	csc := common.NewCache(cacheConfig)
+	csc := common.NewCache(cacheConfig, false)
```

Comment:
Why don't you add the `isScaled` flag into `cacheConfig`?

Reply:
I applied your suggestion. `isScaled` is inside `cacheConfig` from now.